src/fe-gtk/plugin-tray.c

```diff
@@ -509,6 +539,12 @@ tray_menu_cb (GtkWidget *widget, guint button, guint time, gpointer userdata)
 	/* ph may have an invalid context now */
 	hexchat_set_context (ph, hexchat_find_context (ph, NULL, NULL));
 
+	/* close any old menu */
+	if (G_IS_OBJECT (menu))
+	{
+		tray_menu_destroy (menu, NULL);
+	}
+
```

Comment:
With this code I have following issues on OpenIndiana. When one right-cliks hexchat icon in gnome tray several times, hexchat crashes with the following backtrace 

```
Core was generated by hexchat.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  g_type_check_instance_is_fundamentally_a (type_instance=0x826bcd0, 
    fundamental_type=80) at gtype.c:4028
4028	gtype.c: No such file or directory.
[Current thread is 3 (Thread 1 (LWP 1))]
(gdb) bt
#0  0xfeb08f7a in g_type_check_instance_is_fundamentally_a (type_instance=0x826bcd0, fundamental_type=80) at gtype.c:4028
#1  0x0809d2d0 in tray_menu_cb (widget=0x8d63ab8 [GtkStatusIcon], button=3, time=2280341, userdata=0x8d63ab8)
    at /export/home/alp/srcs/oi-userland/components/desktop/hexchat/hexchat-2.10.2/src/fe-gtk/plugin-tray.c:619
#2  0xfe58a459 in _gtk_marshal_VOID__UINT_UINT ()
    at /usr/lib/libgtk-x11-2.0.so.0
#6  0xfeaff504 in <emit signal ??? on instance 0x8d63ab8 [GtkStatusIcon]> (instance=0x8d63ab8, signal_id=230, detail=0) at gsignal.c:3361
    #3  0xfeae4730 in g_closure_invoke (closure=0x8d58158, return_value=0x0, n_param_values=3, param_values=0x8046b00, invocation_hint=0x8046abc)
    at gclosure.c:768
    #4  0xfeaf717b in signal_emit_unlocked_R (node=node@entry=0x87fe190, detail=detail@entry=0, instance=0x8d63ab8, emission_return=0x0, instance_and_params=0x8046b00) at gsignal.c:3549
    #5  0xfeaff342 in g_signal_emit_valist (instance=0x8d63ab8, signal_id=230, detail=0, var_args=0x8046c64 "\270:\326\b\230l\004\bm\300\260\376 n\004\b")
    at gsignal.c:3305
#7  0xfe5fe377 in gtk_status_icon_button_press ()
    at /usr/lib/libgtk-x11-2.0.so.0
#8  0xfe586bd6 in _gtk_marshal_BOOLEAN__BOXED ()
    at /usr/lib/libgtk-x11-2.0.so.0
---Type <return> to continue, or q <return> to quit--- 
#12 0xfeaff504 in <emit signal ??? on instance 0x8254088 [GtkTrayIcon]> (instance=0x8254088, signal_id=33, detail=0) at gsignal.c:3361
    #9  0xfeae4730 in g_closure_invoke (closure=0x81ea508, return_value=0x8046dc0, n_param_values=2, param_values=0x8046e20, invocation_hint=0x8046ddc)
    at gclosure.c:768
    #10 0xfeaf717b in signal_emit_unlocked_R (node=node@entry=0x8133b38, detail=detail@entry=0, instance=0x8254088, emission_return=0x8046ec4, instance_and_params=0x8046e20) at gsignal.c:3549
    #11 0xfeafef32 in g_signal_emit_valist (instance=0x8254088, signal_id=33, detail=0, var_args=0x8046f70 "\234o\004\b\260/ \b\270o\004\b\255\002j\376")
    at gsignal.c:3315
#13 0xfe6a03d8 in gtk_widget_event_internal () at /usr/lib/libgtk-x11-2.0.so.0
#14 0xfe584f8b in gtk_propagate_event () at /usr/lib/libgtk-x11-2.0.so.0
#15 0xfe58545c in gtk_main_do_event () at /usr/lib/libgtk-x11-2.0.so.0
#16 0xfe904528 in gdk_event_dispatch () at /usr/lib/libgdk-x11-2.0.so.0
#17 0xfed25305 in g_main_context_dispatch (context=0x8155ae0) at gmain.c:3122
#18 0xfed25305 in g_main_context_dispatch (context=0x8155ae0) at gmain.c:3737
#19 0xfed25619 in g_main_context_iterate (context=0x8155ae0, block=block@entry=1, dispatch=1, self=<optimized out>) at gmain.c:3808
#20 0xfed25a29 in g_main_loop_run (loop=0x823c350) at gmain.c:4002
#21 0xfe584125 in gtk_main () at /usr/lib/libgtk-x11-2.0.so.0
#22 0x0808479a in fe_main ()
    at /export/home/alp/srcs/oi-userland/components/desktop/hexchat/hexchat-2.10---Type <return> to continue, or q <return> to quit---
.2/src/fe-gtk/fe-gtk.c:320
#23 0x080c32be in main (argc=1, argv=0x8047a30)
    at /export/home/alp/srcs/oi-userland/components/desktop/hexchat/hexchat-2.10.2/src/common/hexchat.c:1137
```